src/utils/utils.js

```diff
@@ -41,8 +41,45 @@ const formatPhone = value => {
   return value.replace(phoneRegex, '$1-$2-$3');
 };
 
+// create address string from address components
+const formatAddress = ({ addressStreet, addressUnit, addressCity, addressZip }) => {
+  const addressArray = [addressStreet, addressUnit, addressCity, `CA ${addressZip}`].filter(
+    Boolean,
+  );
+  return addressArray.join(', ');
+};
+
+// creates Google Maps URL for route and opens it in new window
+const handleNavigateToAddress = donations => {
+  // set PNP address as origin; this will show the origin as Patriots and Paws rather than as just the address on Google Maps
+  const origin = 'Patriots and Paws';
+
+  const addresses = donations.map(donation => formatAddress(donation));
+  const n = addresses.length;
+
+  // set all addresses except last one as waypoint
+  const waypoints = addresses.slice(0, -1).join('|');
+  // set last address as destination
+  const destination = addresses[n - 1];
+
+  const googleMapsUrl = `https://www.google.com/maps/dir/?api=1&origin=${encodeURIComponent(
+    origin,
+  )}&destination=${encodeURIComponent(destination)}&waypoints=${encodeURIComponent(
+    waypoints,
+  )}&travelmode=driving`;
```

Comment:
We recently changed it so the origin will start at the device location, refer [here](https://developers.google.com/maps/documentation/urls/get-started#parameters) in the docs for how to do that. Note that this should be trivially easy, so let us know if you run into any problems at all while trying to do this.